hyperbolic/poincare/Hypercycle.py

```diff
@@ -83,6 +83,16 @@ def intersectionsWithHcycle(self, hcycle2):
             pts.append(Point.fromEuclid(x2, y2))
         except ValueError: pass
         return pts
+    def isPointOnSegment(self, x, y):
+        return self.projShape.isPointOnSegment(x, y)
+    def segmentIntersectionsWithHcycle(self, hcycle2):
+        pts=self.intersectionsWithHcycle(hcycle2)
+        for i, p in enumerate(self.intersectionsWithHcycle(hcycle2)):
+            if self.isPointOnSegment(*p) and hcycle2.isPointOnSegment(*p):
+                continue
+            else:
+                pts=pts.pop(i)
```

Comment:
This is a bug.  `pop` returns the item removed, not the new list.  The list is modified so you don't need to reassign it.

My test code:
```python
import drawSvg as draw
from drawSvg import Drawing
from hyperbolic import euclid
from hyperbolic.poincare.shapes import *

p1 = Point(-.2,-.5)
p2 = Point(.2, .5)
line = Line.fromPoints(*p1, *p2, segment=False)
segment = Line.fromPoints(*p1, *p2, segment=True)
print('Drawn shape:', segment.projShape)

segment2 = Line.fromPoints(-.7, 0, .5, .6, segment=True)
#segment2 = Line.fromPoints(-.7, 0, .2, .6, segment=True)  # <- Triggers bug
print('Drawn shape:', segment2.projShape)

points = [
    segment.startPoint().midpointWith(segment.endPoint(), frac=x)
    for x in [-0.2, 0, 0.2, 0.5, 0.9, 1, 1.3]
]

d = Drawing(2.1, 2.1, origin='center')
d.draw(euclid.shapes.Circle(0, 0, 1), fill='#eee')

d.draw(segment, hwidth=0.15, fill='black')
d.draw(segment2, hwidth=0.15, fill='gray')
for p in points:
    # Blue only if on segment
    fill = 'skyblue' if segment.isPointOnSegment(*p) else 'red'
    d.draw(p, hradius=0.05, fill=fill)
intersects = segment.segmentIntersectionsWithHcycle(segment2)
for p in intersects:
    print('Intersects!')
    d.draw(p, hradius=0.05, fill='white')

d.setRenderSize(w=400)
d
```